### PR TITLE
Feat: add invite route to gateway

### PR DESCRIPTION
**Pull request by Nhsdkk**
1) Add GET: /invites/all/ to get all invites
2) Add GET: /invites/my/invitee/ to get all user's invites as invitee
3) Add GET: /invites/my/author/ to get all user's invites as author
4) Add GET: /invites/{invite_id} to get invite by id
5) Add POST: /invites/ to create invite
6) Add POST: /invites/multiple to create multiple invites
7) Add PUT: /invites/ to update invite
8) Add DELETE: /invites/ to delete invite

### CODE REVIEW

File: gateway/app/models/invite.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    static from_proto(proto)
+        Get invite instance from proto invite
+    to_proto()
+        Get proto invite from invite instance
+
+
+    """
+
+    id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/models/invite.py]: If It's uuid, then set uuid type https://docs.pydantic.dev/2.0/usage/types/uuids/

File: gateway/app/models/invite.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    ]
+    event_id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/models/invite.py]: same

File: gateway/app/models/invite.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    ]
+    author_id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/models/invite.py]: same

File: gateway/app/models/invite.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    ]
+    invitee_id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/models/invite.py]: same

File: gateway/app/routers/invites.py [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+
+from fastapi import APIRouter, Depends, Security
+from pydantic import AfterValidator, BaseModel, Field
+
+router = APIRouter(prefix="/invites", tags=["invites"])
+
+
+class CreateInviteData(BaseModel):
+    invitee_id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/routers/invites.py]: If It's uuid, then set uuid type https://docs.pydantic.dev/2.0/usage/types/uuids/

File: gateway/app/routers/invites.py [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+    ]
+    event_id: Annotated[
+        str, Field("", min_length=1), AfterValidator(str_special_characters_validator)
```
Comment (MaxPlays35) [inline gateway/app/routers/invites.py]: same

**Review by MaxPlays35 — CHANGES_REQUESTED** [review]

**Review by ZixMai — APPROVED** [review]

**Review by MaxPlays35 — CHANGES_REQUESTED** [review]
Use uuid type from pydantic
See https://github.com/pydantic/pydantic/discussions/7023

**Comment by Nhsdkk** [comment]
> Use uuid type from pydantic See [pydantic/pydantic#7023](https://github.com/pydantic/pydantic/discussions/7023)

No need. I have just tested it and the string "id" did not pass, but string uuid passed

**Review by MaxPlays35 — APPROVED** [review]

**Pull request merged**